app/models/payment_confirmation.rb

```diff
@@ -84,6 +84,7 @@ def update_payment_fields(payment, row)
     payment.student_loan_repayment = row["Student Loans"]
     payment.tax = row["Tax"]
     payment.net_pay = row["Net Pay"]
+    payment.gross_pay = row["Gross Value"].to_d - row["Employers NI"].to_d
```

Comment:
If we used the values from the `payment` instance here would we be able to avoid the typecasting?

Reply:
I tried that first, but I got some test failures (NoMethodError on nil) on the spec that checks there are errors if the the values are missing. I could maybe do something like:

```ruby
payment.gross_pay = payment.gross_value - payment.employers_national_insurance if payment.gross_value.present?
```

Reply:
Fair enough. You'd also need to check that `employers_national_insurance` is present also. Probably not worth it.